src/components/TodoApp.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useContext, useState } from 'react';
+import { TodoList } from './TodoList';
+import { TodosFilter } from './TodosFilter';
+import { TodosContext } from './TodosContext';
+import { Todo } from '../types/Todo';
+import { Filter, Status } from '../types/Filters';
+
+const filters: Filter[] = [
+  { href: '/', title: Status.All },
+  { href: '/active', title: Status.Active },
+  { href: '/completed', title: Status.Completed },
+];
+
+export const TodoApp: React.FC = () => {
+  const {
+    todos,
+    setTodos,
+    completedTodos,
+    uncompletedTodos,
+    onToggleAll,
+  } = useContext(TodosContext);
+
+  const [title, setTitle] = useState('');
+  const [filterBy, setFilterBy] = useState(Status.All);
+  const [editing, setEditing] = useState<Todo | null>(null);
+
+  const newTodo: Todo = {
+    id: +new Date(),
```

Comment:
It's not a good idea, as user(s) can add a few todos simultaneously.
Consider using some tool like uuid fo generating an unique id